dwh/indicator_lib.py

```diff
@@ -83,54 +99,37 @@ def calc_TX_PVLS(patient_agg_obs: pandas.DataFrame, VL_code: str,
     end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
   temp_df = patient_agg_obs[(patient_agg_obs['code'] == VL_code)].copy()
   # Note the above copy is used to avoid setting a new column on a slice next:
-  # separate value from date
-  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str.split(query_lib.DATE_VALUE_SEPARATOR,
-                                                                                       expand=True)
-  temp_df['latest_vl_value'] = temp_df.loc[:, 'latest_vl_value'].astype(float)
+  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str\
+    .split(query_lib.DATE_VALUE_SEPARATOR, expand=True)
+  temp_df['latest_vl_value'] = temp_df['latest_vl_value'].astype(float)
 
-  temp_df['sup_VL'] = (temp_df.loc[:, 'latest_vl_value'] < failure_threshold)
-  temp_df['buckets'] = temp_df.apply(
-      lambda x: _agg_buckets(x.birthDate, x.gender, end_date), axis=1)
-  temp_df_exp = temp_df.explode('buckets')
-  temp_df_exp = temp_df_exp.groupby(['sup_VL', 'buckets'], as_index=False)\
-      .count().loc[:, ['sup_VL', 'buckets', 'patientId']]\
-      .rename(columns={'patientId': 'sup_VL_count'})
-  # calculate ratio
-  num_patients = len(temp_df.index)
-  temp_df_exp['sup_VL_ratio'] = temp_df_exp.loc[:, 'sup_VL_count']/num_patients
-  return temp_df_exp
+  temp_df['sup_VL'] = (temp_df['latest_vl_value'] < failure_threshold)
+  temp_df = _gen_counts_and_ratio(temp_df, end_date, 'sup_VL')
+  return temp_df
 
 
-def calc_TX_NEW(patient_agg_obs:  pandas.DataFrame, ARV_plan: str, start_drug: List,
-    end_date_str: str = None) -> pandas.DataFrame:
+def calc_TX_NEW(patient_agg_obs:  pandas.DataFrame, ARV_plan: str,
+                start_drug: List[str], end_date_str: str = None) -> pandas.DataFrame:
   """Calculates TX_NEW indicator with its corresponding disaggregations.
 
   TX_NEW indicator counts the number of adults and children newly enrolled
-  on antiretroviral therapy (ART) within the provided
+  on antiretroviral therapy (ART) prior to the provided end-date
```

Comment:
I don't think this comment is accurate as we use `end-date` only for age calculation in this function. Yes, it is assumed that the provided DataFrame has data up to `end-date` but otherwise we don't do any explicit filtering on date at this stage.